src/grpc_server.cc

```diff
@@ -1340,6 +1340,224 @@ CommonHandler::SetUpAllRequests()
       inference::TraceSettingRequest, inference::TraceSettingResponse>(
       "Trace", 0, OnRegisterTrace, OnExecuteTrace, false /* async */, cq_);
 
+  //
+  //  Log Settings
+  //
+
+  auto OnRegisterLogging =
+      [this](
+          grpc::ServerContext* ctx, inference::LogSettingsRequest* request,
+          grpc::ServerAsyncResponseWriter<inference::LogSettingsResponse>*
+              responder,
+          void* tag) {
+        this->service_->RequestLogSettings(
+            ctx, request, responder, this->cq_, this->cq_, tag);
+      };
+
+  auto OnExecuteLogging = [this](
+                              inference::LogSettingsRequest& request,
+                              inference::LogSettingsResponse* response,
+                              grpc::Status* status) {
+
+#ifdef TRITON_ENABLE_LOGGING
+    TRITONSERVER_Error* err = nullptr;
+    std::string log_file_path = LOG_OUT_FILE;
+    bool log_info_status = LOG_INFO_IS_ON;
+    bool log_warn_status = LOG_WARNING_IS_ON;
+    bool log_error_status = LOG_ERROR_IS_ON;
+    uint32_t verbose_level = LOG_VERBOSE_LEVEL;
+    triton::common::Logger::Format log_format_final = LOG_FORMAT;
+    std::string log_format_parse = LOG_FORMAT_STRING;
+    // Update log settings
+    // Server and Core repos do not have the same Logger object
+    // Each update must be applied to both server and core repo versions
+    if (!request.settings().empty()) {
+      {
+        static std::string setting_name = "log_file";
+        auto it = request.settings().find(setting_name);
+        if (it != request.settings().end()) {
+          const auto& log_param = it->second;
+          if (log_param.parameter_choice_case() !=
+              inference::LogSettingsRequest_SettingValue::ParameterChoiceCase::
+                  kStringParam) {
+            err = TRITONSERVER_ErrorNew(
+                TRITONSERVER_ERROR_INVALID_ARG,
+                (std::string("expect string for '") + setting_name + "'")
+                    .c_str());
+            GOTO_IF_ERR(err, earlyexit);
+          } else {
+            // Set new settings in server then in core
+            log_file_path = it->second.string_param();
+            LOG_SET_OUT_FILE(log_file_path);
+            FAIL_IF_ERR(
+                TRITONSERVER_ServerOptionsSetLogOutFile(
+                    NULL, log_file_path.c_str()),
```

Comment:
`nullptr`, should add comment here and below on passing `nullptr` is okay because we know the implementation detail that the update will be applied to global object so no pointer to object needs to be provided to this API call.